src/components/AppNavigator/AppNavigator.tsx

```diff
@@ -1,100 +1,111 @@
-import { useContext } from 'react';
-import { Outlet, Link, useLocation } from 'react-router-dom';
+import { useContext, useEffect } from 'react';
+import { Link as RouterLink, useLocation, useNavigate } from 'react-router-dom';
+import Link from '@mui/material/Link';
 import Step from '@mui/material/Step';
 import Stepper from '@mui/material/Stepper';
-import StepContent from '@mui/material/StepContent';
 import StepLabel from '@mui/material/StepLabel';
+import Typography from '@mui/material/Typography';
 import FormScreens from 'constants/formScreens';
 import routes from 'constants/routes';
 import WizardFormContext from 'contexts/WizardFormContext';
 import Entries from 'types/entries';
 import { FormValuesType } from 'types/formTypes';
 
-interface MenuItemType {
-  step: FormScreens;
-  isCompleted: boolean;
-  url: string;
+interface Props {
+  className?: string;
 }
 
-export default function AppNavigator() {
-  const { formValues } = useContext(WizardFormContext);
-  const location = useLocation();
+export default function AppNavigator({ className }: Props) {
+  const { formValues, firstUncompletedStep } = useContext(WizardFormContext);
+  const { pathname } = useLocation();
+  const navigate = useNavigate();
+  // https://www.charpeni.com/blog/properly-type-object-keys-and-object-entries
+  const formValuesEntries = Object.entries(
+    formValues,
+  ) as Entries<FormValuesType>;
+  type FormValuesKeys = keyof typeof formValues;
```

Comment:
I don't like that our TS definitions are mixed with logic in the component. But I need to think how to improve it

Reply:
You can replace `typeof formValues` by `FormValuesType` and move these things to `formTypes`